Doc/library/functions.rst

```diff
@@ -496,44 +496,38 @@ are always available.  They are listed here in alphabetical order.
 
 .. function:: eval(expression[, globals[, locals]])
 
-   The arguments are a string and optional globals and locals.  If provided,
-   *globals* must be a dictionary.  If provided, *locals* can be any mapping
-   object.
-
-   The *expression* argument is parsed and evaluated as a Python expression
-   (technically speaking, a condition list) using the *globals* and *locals*
-   dictionaries as global and local namespace.  If the *globals* dictionary is
-   present and does not contain a value for the key ``__builtins__``, a
-   reference to the dictionary of the built-in module :mod:`builtins` is
-   inserted under that key before *expression* is parsed.  That way you can
-   control what builtins are available to the executed code by inserting your
-   own ``__builtins__`` dictionary into *globals* before passing it to
-   :func:`eval`.  If the *locals* dictionary is omitted it defaults to the
-   *globals* dictionary.  If both dictionaries are omitted, the expression is
-   executed with the *globals* and *locals* in the environment where
-   :func:`eval` is called.  Note, *eval()* does not have access to the
-   :term:`nested scopes <nested scope>` (non-locals) in the enclosing
-   environment.
-
-   The return value is the result of
-   the evaluated expression. Syntax errors are reported as exceptions.  Example:
+   This function supports the dynamic evaluation of Python expression.  The
+   first argument can be a string or a code object.  The optional arguments
+   specify the globals and locals respectively.  If provided, *globals* must be
+   a dictionary.  If provided, *locals* can be any mapping object.
+
+   In the most common case, the *expression* argument is a string, and it is
+   parsed and evaluated as a Python expression (see the section
+   ":ref:`expression-input`" in the Language Reference).  The leading and
```

Comment:
Sorry, I don't remember if this is valid/invalid inline markup. Some roles require whitespace around them. See this to learn more:
https://devguide.python.org/documenting/#inline-markup

So it might be ``"\ :ref:`expression-input`\ "``. However, if you ran ``./make html`` and all looks fine on your computer, then I'm wrong and please ignore this comment :).

Reply:
Hi, I think this works fine (verified with the Sphinx setup created by "make venv" and "make html") -- unlike inline markup, the `ref` cross-linking markup seem to be Ok near punctuations.